File: scratch/fix_json.js
const fs = require('fs');

const content = fs.readFileSync('../data/niches.json', 'utf8');
const lines = content.split('\n');
const headLines = [];
let inConflict = false;
let keep = true;

for (const line of lines) {
    if (line.startsWith('<<<<<<<')) {
        inConflict = true;
        keep = true;
        continue;
    }
    if (line.startsWith('=======')) {
        keep = false;
        continue;
    }
    if (line.startsWith('>>>>>>>')) {
        inConflict = false;
        keep = true;
        continue;
    }
    if (keep) {
        headLines.push(line);
    }
}

const headJson = headLines.join('\n');
try {
    const data = JSON.parse(headJson);
    fs.writeFileSync('../data.js', 'const localDatabase = ' + JSON.stringify(data, null, 2) + ';\n');
    console.log("Success HEAD");
} catch (e) {
    console.log("Failed HEAD parse:", e.message);
    // Write out the raw JSON to check where it failed
    fs.writeFileSync('failed.json', headJson);
}

File: scratch/fix_json.py
import json
import re

with open('../data/niches.json', 'r', encoding='utf-8') as f:
    content = f.read()

lines = content.split('\n')
head_lines = []
in_conflict = False
keep = True
for line in lines:
    if line.startswith('<<<<<<<'):
        in_conflict = True
        keep = True
        continue
    if line.startswith('======='):
        keep = False
        continue
    if line.startswith('>>>>>>>'):
        in_conflict = False
        keep = True
        continue
    
    if keep:
        head_lines.append(line)

head_json = '\n'.join(head_lines)
try:
    data = json.loads(head_json)
    with open('../data.js', 'w', encoding='utf-8') as out:
        out.write('const localDatabase = ' + json.dumps(data, indent=2) + ';\n')
    print("Success HEAD")
except Exception as e:
    print("Failed HEAD parse:", e)
